lib/class-wp-theme-json-gutenberg.php

```diff
@@ -243,6 +244,16 @@ class WP_Theme_JSON_Gutenberg {
 		'text-transform'             => array( 'typography', 'textTransform' ),
 	);
 
+	/**
+	 * Metadata for style properties that need to use the duotone selector.
+	 *
+	 * Each element is a direct mapping from the CSS property name to the
+	 * path to the value in theme.json & block attributes.
+	 */
+	const DUOTONE_PROPERTIES_METADATA = array(
+		'filter' => array( 'color', 'duotone' ),
+	);
```

Comment:
@oandregal I see this as a stepping stone to eventually get us to a better way to configure styles that need different selectors. Since these selectors are still experimental (`__experimentalSelector` and `__experimentalDuotone`), I didn't want to worry too much about the final API yet.

Reply:
This is nice but has the drawback to have to consider the two arrays in other methods (`remove_insecure_styles`). I've pushed a tweak and commenting at https://github.com/WordPress/gutenberg/pull/34667#discussion_r713179970

Reply:
I've reverted the change I had done, so we need to make this work with `remove_insecure_styles`. There're some tests we can update to consider duotone at `WP_Theme_JSON_Gutenberg_Test.test_remove_insecure_properties_removes_unsafe_styles`